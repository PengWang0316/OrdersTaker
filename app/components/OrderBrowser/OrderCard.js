import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Card, CardMedia, CardContent, Typography, IconButton, Tooltip } from '@material-ui/core';
import { DeleteForever, Link } from '@material-ui/icons';
import { red } from '@material-ui/core/colors';

import OrderBrowserCardHeader from './OrderBrowserCardHeader';
import OrdersPageContext from '../../contexts/OrdersPageContext';

const styles = {
  card: {
    minWidth: 300,
    maxWidth: 500,
    margin: '10px 10px',
    '&:hover': {
      cursor: 'pointer'
    }
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  cardContent: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between'
  },
  iconButton: {
    width: 30,
    height: 30
  },
  deleteButton: {
    color: red[400]
  },
  icon: {
    width: 20,
    height: 20
  }
};

export const OrderCard = ({ order, classes, menuItems }) => {
  const itemsKey = Object.keys(order.items);
  return (
    <OrdersPageContext.Consumer>
      {({ handleShowDetail, handleRmoveUnloginOrder, handleLinkOrder }) => (
        <Card className={classes.card}>
          <OrderBrowserCardHeader order={order} onClick={() => handleShowDetail(order)} />
          <CardMedia className={classes.media} image={menuItems[itemsKey[0]].photo} title={menuItems[itemsKey[0]].name} onClick={() => handleShowDetail(order)} />
          <CardContent className={classes.cardContent}>
            <div>
              <Typography className={classes.title} color="textSecondary">This order has {itemsKey.length} items</Typography>
              <Typography color="textSecondary">Total Price: ${order.totalPrice}</Typography>
            </div>
            {!order.userId && (
              <div>
                <Tooltip title="Remove this order from this device">
                  <IconButton onClick={() => handleRmoveUnloginOrder(order._id)} className={`${classes.iconButton} ${classes.deleteButton}`}><DeleteForever className={classes.icon} /></IconButton>
                </Tooltip>
                <Tooltip title="Link this order to your account">
                  <IconButton onClick={() => handleLinkOrder(order._id)} color="primary" className={classes.iconButton}><Link className={classes.icon} /></IconButton>
                </Tooltip>
              </div>
            )}
          </CardContent>
        </Card>
      )}
    </OrdersPageContext.Consumer>
  );
};
OrderCard.propTypes = {
  order: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
  menuItems: PropTypes.object.isRequired
};
export default withStyles(styles)(OrderCard);
