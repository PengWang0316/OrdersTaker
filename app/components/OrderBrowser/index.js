import OrderBrowser from './OrderBrowser';

export default OrderBrowser;
