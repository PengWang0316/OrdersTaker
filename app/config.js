export const JWT_MESSAGE = 'OrdersTaker_JWT';
export const ADMINISTRATOR_ROLE = 1;
export const SUPER_USER_ROLE = 2;
export const NORMAL_USER_ROLE = 3;
// export const BASE_URL = 'https://orders-taker.glitch.me'; // Development server
// export const BASE_URL = 'https://orderstaker.kevin-project.com:8081'; // Development server on the AWS
export const BASE_URL = 'https://orderstaker.kevin-project.com:8080'; // Production server
// export const SOCKETIO_URL = 'https://orderstaker.kevin-project.com:3001';
export const SOCKETIO_URL = 'https://orderstaker.kevin-project.com:3000'; // Production server

export const CLOUDINARY_UPLOAD_PRESET = 'OrdersTakerWebApp'; // The preset of Cloudinary.
export const LOGIN_CALLBACK_URL = 'OrdersTaker_loginCallbackUrl';

export const ITEM_ONE_PRICE_KEY = '_onePrice';
export const MAX_ORDER_AMOUNT = 6; // How many orders will show in the one page
export const ORDER_STATUS_FINISHED = 'Finished';
export const ORDER_STATUS_RECEIVED = 'Received';
export const ORDER_STATUS_PREPARING = 'Preparing';

/* Lazy image loading */
export const LAZY_IMAGE_CLASS = 'lazyImage';
export const IMAGE_PLACEHOLDER_URL = 'https://res.cloudinary.com/orderstaker/image/upload/v1536616770/others/PlaceHolder.png';

/* URLS */
export const HOME_PAGE_URL = '/';
export const LOGIN_REDIRECT_RUL = '/loginRedirect';
export const CART_PAGE_URL = '/cartPage';
export const ORDERS_PAGE_URL = '/ordersPage';
export const KITHEN_INTERFACE_PAGE_URL = '/kithenInterfacePage';

/* Socket IO events */
export const SOCKETIO_EVENT_ADD_NEW_ORDER = 'socketioEventAddNewOrder';
export const SOCKETIO_EVENT_UPDATE_ORDER_ITEM = 'socketioEventUpdateOrderItem';
