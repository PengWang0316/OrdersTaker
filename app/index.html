<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="A simply order taking system for restaurants">
    <meta name="keywords" content="Order taker restaurant free">
    <!-- Add manifest -->
    <link rel="manifest" href="/manifest.json">
    <!-- Tell the browser it's a PWA -->
    <meta name="mobile-web-app-capable" content="yes">
    <!-- Tell iOS it's a PWA -->
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="theme-color" content="#6d9cb5"/>
    <link rel="icon" href="/favicon.png" >
    <!-- Latest compiled and minified CSS -->

    <title>Orders Taker - A simply order taking system for restaurants</title>
  </head>
  <body>
    <div id="app"></div>
    <noscript>This site requires JavaScript to work. Please change to a browser that supports JavaScript or turn on JavaScript in the setting.</noscript>
    <!-- Latest compiled and minified JavaScript -->
  </body>
</html>
