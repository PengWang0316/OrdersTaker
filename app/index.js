import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { AppContainer } from 'react-hot-loader';
import { PersistGate } from 'redux-persist/integration/react';

import App from './components/App';
import configureStore from './store/ConfigureStore';
import './styles/index.global.css';
import registerServiceWorker from './registerServiceWorker';

const { store, persistor } = configureStore();

// ReactDOM.render(
//   <Provider store={store}>
//     <App />
//   </Provider>,
//   document.getElementById('app')
// );
const render = Component => {
  ReactDOM.render(
    <AppContainer>
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <Component />
        </PersistGate>
      </Provider>
    </AppContainer>,
    document.getElementById('app'),
  );
};
render(App);

// Webpack Hot Module Replacement API
if (module.hot) module.hot.accept('./components/App', () => render(App));

// Registering a service worker in the production enviroment.
registerServiceWorker();
