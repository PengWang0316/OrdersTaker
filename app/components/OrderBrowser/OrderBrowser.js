import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import Pagination from '@kevinwang0316/react-materialui-pagination';
// import Pagination from '../Pagination';

import OrderCard from './OrderCard';
import { MAX_ORDER_AMOUNT } from '../../config';

const styles = {
  root: {
    width: '90%',
    margin: '40px auto'
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold'
  },
  cardDiv: {
    display: 'flex',
    flexWrap: 'wrap'
  }
};

export const OrderBrowser = ({
  orders, classes, title, menuItems, offset, totalOrder, onPageNumberClick
}) => (
  <div className={classes.root}>
    <Typography color="primary" className={classes.title}>{title}</Typography>
    {Object.keys(menuItems).length !== 0 && (
      <div className={classes.cardDiv}>
        {orders.map(order => <OrderCard order={order} menuItems={menuItems} key={order._id} />)}
      </div>
    )}
    <Pagination
      offset={offset}
      limit={MAX_ORDER_AMOUNT}
      total={totalOrder}
      onClick={onPageNumberClick}
    />
  </div>
);
OrderBrowser.prototype = {
  orders: PropTypes.array.isRequired,
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  menuItems: PropTypes.object.isRequired,
  onPageNumberClick: PropTypes.func.isRequired,
  offset: PropTypes.number.isRequired,
  totalOrder: PropTypes.number.isRequired
};
/* istanbul ignore next */
const mapStateToProps = state => ({
  menuItems: state.menuItems
});
export default connect(mapStateToProps, null)(withStyles(styles)(OrderBrowser));
