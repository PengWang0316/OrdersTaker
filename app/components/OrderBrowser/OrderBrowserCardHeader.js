import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { CardHeader, Tooltip } from '@material-ui/core';
import { lightGreen, yellow, grey, red } from '@material-ui/core/colors';
import { CheckCircle, OfflinePin, WatchLater, Help } from '@material-ui/icons';

import { ORDER_STATUS_FINISHED, ORDER_STATUS_RECEIVED, ORDER_STATUS_PREPARING } from '../../config';

const avatarIcon = {
  width: 40,
  height: 40
};

const styles = {
  checkCircle: {
    ...avatarIcon,
    color: lightGreen[700]
  },
  offlinPin: {
    ...avatarIcon,
    color: yellow[800]
  },
  watchLater: {
    ...avatarIcon,
    color: grey[500]
  },
  unknow: {
    ...avatarIcon,
    color: red[500]
  }
};

export const OrderBrowserCardHeader = ({ order, classes, ...rest }) => {
  const orderDate = new Date(order.dateStamp);
  let avatarElement;
  switch (order.status) {
    case ORDER_STATUS_FINISHED:
      avatarElement = (
        <Tooltip title="Finished" placement="right-end">
          <CheckCircle className={classes.checkCircle} />
        </Tooltip>
      );
      break;
    case ORDER_STATUS_RECEIVED:
      avatarElement = (
        <Tooltip title="Received" placement="right-end">
          <OfflinePin className={classes.offlinPin} />
        </Tooltip>
      );
      break;
    case ORDER_STATUS_PREPARING:
      avatarElement = (
        <Tooltip title="Preparing" placement="right-end">
          <WatchLater className={classes.watchLater} />
        </Tooltip>
      );
      break;
    default:
      avatarElement = (
        <Tooltip title="Unknow Status" placement="right-end">
          <Help className={classes.unknow} />
        </Tooltip>
      );
      break;
  }
  return (
    <CardHeader
      avatar={avatarElement}
      title={orderDate.toDateString()}
      subheader={orderDate.toLocaleTimeString()}
      {...rest}
    />
  );
};
OrderBrowserCardHeader.propTypes = {
  order: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired
};
export default withStyles(styles)(OrderBrowserCardHeader);
