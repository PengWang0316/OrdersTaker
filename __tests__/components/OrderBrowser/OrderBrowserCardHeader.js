import React from 'react';
import renderer from 'react-test-renderer';

import { OrderBrowserCardHeader } from '../../../app/components/OrderBrowser/OrderBrowserCardHeader';
import { ORDER_STATUS_FINISHED, ORDER_STATUS_RECEIVED, ORDER_STATUS_PREPARING } from '../../../app/config';

jest.mock('@material-ui/core/CardHeader', () => 'CardHeader');
jest.mock('@material-ui/core/Tooltip', () => 'Tooltip');
jest.mock('@material-ui/icons/CheckCircle', () => 'CheckCircle');
jest.mock('@material-ui/icons/OfflinePin', () => 'OfflinePin');
jest.mock('@material-ui/icons/WatchLater', () => 'WatchLater');
jest.mock('@material-ui/icons/Help', () => 'Help');

describe('OrderBrowserCardHeader', () => {
  const defaultProps = {
    classes: {
      checkCircle: 'checkCircle',
      offlinPin: 'offlinPin',
      watchLater: 'watchLater',
      unknow: 'unkonw'
    },
    order: {
      dateStamp: '2018-07-24T00:20:48.033Z',
      status: ORDER_STATUS_FINISHED
    },
    onClick: jest.fn()
  };

  test('Snapshot ORDER_STATUS_FINISHED', () => expect(renderer.create(<OrderBrowserCardHeader {...defaultProps} />).toJSON()).toMatchSnapshot());
  test('Snapshot ORDER_STATUS_RECEIVED', () => expect(renderer.create(<OrderBrowserCardHeader {...{ ...defaultProps, order: { dateStamp: '2018-07-24T00:20:48.033Z', status: ORDER_STATUS_RECEIVED } }} />).toJSON()).toMatchSnapshot());
  test('Snapshot ORDER_STATUS_PREPARING', () => expect(renderer.create(<OrderBrowserCardHeader {...{ ...defaultProps, order: { dateStamp: '2018-07-24T00:20:48.033Z', status: ORDER_STATUS_PREPARING } }} />).toJSON()).toMatchSnapshot());
  test('Snapshot unknow status', () => expect(renderer.create(<OrderBrowserCardHeader {...{ ...defaultProps, order: { dateStamp: '2018-07-24T00:20:48.033Z', status: '' } }} />).toJSON()).toMatchSnapshot());
});
