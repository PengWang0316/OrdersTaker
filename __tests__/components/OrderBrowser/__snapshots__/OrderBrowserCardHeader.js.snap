// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderBrowserCardHeader Snapshot ORDER_STATUS_FINISHED 1`] = `
<CardHeader
  avatar={
    <Tooltip
      placement="right-end"
      title="Finished"
    >
      <CheckCircle
        className="checkCircle"
      />
    </Tooltip>
  }
  onClick={[MockFunction]}
  subheader="17:20:48"
  title="Mon Jul 23 2018"
/>
`;

exports[`OrderBrowserCardHeader Snapshot ORDER_STATUS_PREPARING 1`] = `
<CardHeader
  avatar={
    <Tooltip
      placement="right-end"
      title="Preparing"
    >
      <WatchLater
        className="watchLater"
      />
    </Tooltip>
  }
  onClick={[MockFunction]}
  subheader="17:20:48"
  title="Mon Jul 23 2018"
/>
`;

exports[`OrderBrowserCardHeader Snapshot ORDER_STATUS_RECEIVED 1`] = `
<CardHeader
  avatar={
    <Tooltip
      placement="right-end"
      title="Received"
    >
      <OfflinePin
        className="offlinPin"
      />
    </Tooltip>
  }
  onClick={[MockFunction]}
  subheader="17:20:48"
  title="Mon Jul 23 2018"
/>
`;

exports[`OrderBrowserCardHeader Snapshot unknow status 1`] = `
<CardHeader
  avatar={
    <Tooltip
      placement="right-end"
      title="Unknow Status"
    >
      <Help
        className="unkonw"
      />
    </Tooltip>
  }
  onClick={[MockFunction]}
  subheader="17:20:48"
  title="Mon Jul 23 2018"
/>
`;
